core/src/network/handler/mod.rs

```diff
@@ -253,53 +262,73 @@ fn get_content_aspect(
+        .find(|ref chain_header| chain_header.address() == *entry_address)
+        .map(|h| (h, true))
+        .or_else(|| {
+            state
+                .agent()
+                .iter_chain()
+                .find(|ref chain_header| chain_header.entry_address() == entry_address)
+                .map(|h| (h, false))
+        });
 
     // If we have found a header for the requested entry in the chain...
-    let maybe_entry_with_header = if let Some(header) = maybe_chain_header {
-        // ... we can just get the content from the chain CAS
-        Some(EntryWithHeader {
-            entry: get_entry_from_cas(&state.agent().chain_store().content_storage(), header.entry_address())?
+    let maybe_entry_with_header = match maybe_chain_header {
+        Some((header, true)) => Some(create_entry_with_header_for_header(&*state, &header)?),
+        Some((header, false)) => {
+            // ... we can just get the content from the chain CAS
+            Some(EntryWithHeader {
+                entry: get_entry_from_cas(
+                    &state.agent().chain_store().content_storage(),
+                    header.entry_address(),
+                )?
                 .expect("Could not find entry in chain CAS, but header is chain"),
-            header
-        })
-    } else {
-        // ... but if we didn't author that entry, let's see if we have it in the DHT cas:
-        if let Some(entry) = get_entry_from_cas(&state.dht().content_storage(), entry_address)? {
-            // If we have it in the DHT cas that's good,
-            // but then we have to get the header like this:
-            let headers = context
-                .state()
-                .expect("Could not get state for handle_fetch_entry")
-                .get_headers(entry_address.clone())
-                .map_err(|error| {
-                    let err_message = format!(
-                        "net/fetch/get_content_aspect: Error trying to get headers {:?}",
-                        error
+                header,
+            })
+        }
+        None => {
+            // ... but if we didn't author that entry, let's see if we have it in the DHT cas:
+            if let Some(entry) = get_entry_from_cas(&state.dht().content_storage(), entry_address)?
+            {
+                // If we have it in the DHT cas that's good,
+                // but then we have to get the header like this:
+                let headers = context
+                    .state()
+                    .expect("Could not get state for handle_fetch_entry")
+                    .get_headers(entry_address.clone())
+                    .map_err(|error| {
+                        let err_message = format!(
+                            "net/fetch/get_content_aspect: Error trying to get headers {:?}",
+                            error
+                        );
+                        log_error!(context, "{}", err_message.clone());
```

Comment:
No need to clone here. Format should take by reference.
```suggestion
                        log_error!(context, "{}", err_message);
```